### PR TITLE
30 feat style login page

### LINKED ISSUE #30: FEAT: style login page

### PR DESCRIPTION
**Pull request by vitor-afonso**
### PR description
- Adds React-hook-form to the login form
- Creates generic `InputText` component to be reused in all forms
- Updates `APP.constants` to have input names used in forms
- Adds some styling to `Input page`
- Creates generic `Button` component 

Screenshot 2023-07-15 at 13 07 51

### CODE REVIEW

File: src/components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+interface Props {
+  innerText: string;
+  clickHandler?(): void;
+}
+const Button = ({ innerText, clickHandler }: Props) => {
+  return (
+    <button
+      type='submit'
```
Comment (gustavosf2003) [inline src/components/Button.tsx]: Don't need to use `type="submit"`

File: src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      className='w-full flex justify-center py-2 px-4 border border-transparent rounded-3xl shadow-sm text-md font-medium text-gray-300 hover:text-slate-700 bg-slate-500 hover:bg-slate-300 focus:outline-none focus:ring-1 focus:ring-offset-2 focus:ring-slate-500 h-10 duration-200 capitalize'
+      onClick={clickHandler}
+    >
+      {innerText}
```
Comment (gustavosf2003) [inline src/components/Button.tsx]: Pass children as props instead of inner text

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+interface Props {
+  innerText: string;
+  clickHandler?(): void;
+}
+const Button = ({ innerText, clickHandler }: Props) => {
+  return (
+    <button
+      type='submit'
+      className='w-full flex justify-center py-2 px-4 border border-transparent rounded-3xl shadow-sm text-md font-medium text-gray-300 hover:text-slate-700 bg-slate-500 hover:bg-slate-300 focus:outline-none focus:ring-1 focus:ring-offset-2 focus:ring-slate-500 h-10 duration-200 capitalize'
+      onClick={clickHandler}
```
Comment (gustavosf2003) [inline src/components/Button.tsx:12]: use {...rest} instead of writing button props

File: src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    >
+      {innerText}
+    </button>
+  );
+};
+
+export default Button;
```
Comment (gustavosf2003) [inline src/components/Button.tsx:19]: https://github.com/gustavosf2003/fresh-market/blob/main/admin/src/components/Button/Index.tsx

File: src/components/InputText.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  errors: any;
+  inputName: string;
+  inputType?: string;
+}
+
+const InputText = ({ register, errors, inputName, inputType }: Props) => {
+  return (
+    <div>
+      <label>
+        <p className='text-gray-300 text-lg capitalize'>{inputName}</p>
```
Comment (gustavosf2003) [inline src/components/InputText.tsx]: don't need to use a` <p>` inside label

**Review by gustavosf2003 — CHANGES_REQUESTED** [review]

File: src/components/InputText.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+interface Props {
+  register(name: string, {}): any;
```
Comment (zandrastr) [inline src/components/InputText.tsx]: I dont think you should use `any`

File: src/components/InputText.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  errors: any;
```
Comment (zandrastr) [inline src/components/InputText.tsx:5]: Same here, type `any` should be avoided

**Review by zandrastr — CHANGES_REQUESTED** [review]
Check comments

File: src/components/Button.tsx (hunk 3 above)
Comment (vitor-afonso) [reply src/components/Button.tsx:12]: That was cool, i did not know that even the function would be recognized as an `onClick` handler! Thanks 🙌

**Pull request merged**